File: Assets/ShootingRPGKit/CsharpExample/ScriptCS/EnemyScript/DetectCollider.cs
using UnityEngine;
using System.Collections;

[RequireComponent (typeof(Rigidbody))]
public class DetectCollider : MonoBehaviour {
	public Transform master;
	private AIenemy ai;
	
	void Start (){
		if(!master){
			master = transform.root;
		}
		ai = master.GetComponent<AIenemy>();
		gameObject.layer = 2;
		GetComponent<Rigidbody>().isKinematic = true;
		GetComponent<Collider>().isTrigger = true;
	}
	
	void OnTriggerEnter (Collider other){
		if(ai.followState == AIState.Moving || ai.followState == AIState.Pausing){
			return;
		}
		if(other.gameObject.tag == "Player" || other.gameObject.tag == "Ally"){
			ai.followTarget = other.transform;
			ai.followState = AIState.Moving;
		}
	}
}

File: Assets/ShootingRPGKit/Scripts/EnemyScript/DetectCollider.js
#pragma strict
var master : Transform;
private var ai : AIenemy;

function Start () {
	if(!master){
		master = transform.root;
	}
	ai = master.GetComponent(AIenemy);
	gameObject.layer = 2;
	GetComponent(Rigidbody).isKinematic = true;
	GetComponent.<Collider>().isTrigger = true;
}

function OnTriggerEnter (other : Collider) {
	if(ai.followState == AIState.Moving || ai.followState == AIState.Pausing){
		return;
	}
  	if (other.gameObject.tag == "Player" || other.gameObject.tag == "Ally") {
		ai.followState = AIState.Moving;
	}
}

@script RequireComponent(Rigidbody)